backend/repositories/ContentPublisherRepository.ts

```diff
@@ -0,0 +1,64 @@
+import { Client as ContentPublisherClient } from "../types/openapi-content-publisher.js";
+import type { Components, Paths } from "../types/openapi-content-publisher.js";
+import OpenAPIClientAxios from "openapi-client-axios";
+export class ContentPublisherRepository {
+  private static instance: ContentPublisherRepository;
+  private _client: ContentPublisherClient;
+
+  private constructor() {}
+
+  public static async getInstance(): Promise<ContentPublisherRepository> {
+    if (!ContentPublisherRepository.instance) {
+      ContentPublisherRepository.instance = new ContentPublisherRepository();
+      await ContentPublisherRepository.instance.connect();
+    }
+    return ContentPublisherRepository.instance;
+  }
+
+  private async connect() {
+    if (this._client === undefined) {
+      const api = new OpenAPIClientAxios({
+        definition: "../../openapi-content-publisher.json",
+      });
+
+      this.client = await api.init<ContentPublisherClient>();
+    }
+  }
+  private set client(api: ContentPublisherClient) {
+    this._client = api;
+  }
+
+  private get client() {
+    if (this._client === undefined) {
+      throw new Error("API not initialized");
+    }
+
+    return this._client;
+  }
+
+  public async postBroadcast(
+    msaId: string,
+    data: Components.Schemas.BroadcastDto,
```

Comment:
Semantic, but can we just import the needed schemas or unwrap them like `const {BroadcastDto, AnnouncementResponseDto, ReplyDto} = Components.Schemas;`

Reply:
These are types so we are unable to desconstruct them in that way. But your right its not great for reading. I updated the code to make it easier to read and not have to go through the namespace.